c_glib/arrow-glib/input-stream.cpp

```diff
@@ -516,6 +516,10 @@ namespace garrow {
       return input_stream_;
     }
 
+    bool closed() const override {
+      return !!g_input_stream_is_closed(input_stream_);
```

Comment:
@kou Does this look fine to you?

Reply:
That's fine except `!!` is needless.

Reply:
Won't the implicit cast from `gboolean` to `bool` emit a warning otherwise?

Reply:
Umm. I can't reproduce the warning with `g++ (Debian 8.2.0-8) 8.2.0` without `!!`.
Anyway, I understand what you want to care. it's OK to me that we use explicit cast here but I want to use `static_cast<bool>(...)` instead `!!...` in GLib bindings.